app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt

```diff
@@ -114,4 +126,75 @@ open class TripsViewModel(
   }
+
+  open fun addActivityToTrip(activity: Activity) {
+    if (selectedTrip.value != null) {
+      val trip = selectedTrip.value!!
+      val updatedTrip = trip.copy(activities = trip.activities + activity)
+      updateTrip(
+          updatedTrip,
+          onSuccess = {
+            /*
+                This is a trick to force a recompose, because the reference wouldn't
+                change and update the UI.
+            */
+            selectTrip(Trip())
+            selectTrip(updatedTrip)
+          })
+    }
+  }
+
+  // ****************************************************************************************************
+  // AI assistant
+  // ****************************************************************************************************
+  private val generativeModel =
+      GenerativeModel(
+          modelName = "gemini-1.5-flash",
+          apiKey = BuildConfig.GEMINI_API_KEY,
+          generationConfig =
+              generationConfig {
+                responseMimeType = "application/json"
+                responseSchema =
+                    Schema(
+                        name = "activities",
+                        description = "List of activities",
+                        type = FunctionType.ARRAY,
+                        items =
+                            Schema(
+                                name = "activity",
+                                description = "An activity",
+                                type = FunctionType.OBJECT,
+                                properties =
+                                    mapOf(
+                                        "title" to
+                                            Schema(
+                                                name = "title",
+                                                description = "Title of the activity",
+                                                type = FunctionType.STRING),
+                                        "description" to
+                                            Schema(
+                                                name = "description",
+                                                description = "Description of the activity",
+                                                type = FunctionType.STRING)),
+                                required = listOf("title", "description")))
+              })
+
+  fun sendActivitiesPrompt(trip: Trip, prompt: String) {
+    _uiState.value = UiState.Loading
+
+    viewModelScope.launch(Dispatchers.IO) {
+      try {
+        val response =
+            generativeModel.generateContent(
+                "List a few popular specific activities to do on a trip called ${trip.name} and with the following prompt: $prompt")
+        response.text?.let { outputContent -> _uiState.value = UiState.Success(outputContent) }
+      } catch (e: Exception) {
+        _uiState.value = UiState.Error(e.localizedMessage ?: "")
```

Comment:
Not really a problem right now, but consider replacing the null string fallback with a more descriptive and user-friendly error message.

Reply:
Great, thank you very much!